src/ufuncs.c

```diff
@@ -222,21 +236,25 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
                      npy_intp *steps, void *data)
 {
     FuncGEOS_Y_Y *func = (FuncGEOS_Y_Y *)data;
-    void *context_handle = geos_context[0];
     GEOSGeometry *in1, *ret_ptr;
 
+    GEOS_INIT;
+
     UNARY_LOOP {
         /* get the geometry: return on error */
-        if (!get_geom(*(GeometryObject **)ip1, &in1)) { return; }
+        if (!get_geom(*(GeometryObject **)ip1, &in1)) { errstate = PGERR_NOT_A_GEOMETRY; goto finish; }
         if (in1 == NULL) {
-            /* in case of a missing value: return NULL (NaG) */
+            /* in case of a missing value: return NULL (None) */
             ret_ptr = NULL;
         } else {
-            ret_ptr = func(context_handle, in1);
-            /* trust that GEOS calls HandleGEOSError on error */
+            ret_ptr = func(ctx, in1);
+            if ((ret_ptr == NULL) && (last_error[0] != 0)) { errstate = PGERR_GEOS_EXCEPTION; goto finish; }
```

Comment:
why is the extra check of `last_error[0]` needed in this case? (some times it is done, and sometimes not)
(maybe add a comment about it)